MapleServer2/Servers/Game/FieldManager.cs

```diff
@@ -612,5 +613,15 @@ private Task StartSpawnTimer(IFieldObject<MobSpawn> mobSpawn)
                 }
             });
         }
+
+        public int Pin()
```

Comment:
You need to use `Interlocked.Increment` and `Interlocked.Decrement` for thread safety.